k8s/src/main/scala/io/buoyant/k8s/resources.scala

```diff
@@ -28,7 +30,12 @@ trait Resource extends Closable {
  * @tparam O The parent type for all [Object]s served by this version.
  */
 private[k8s] trait Version[O <: KubeObject] extends Resource {
-  val path = s"/$group/$version"
+  // alex, I know you don't like lazy-evaluation, but i thought it was
+  // reasonable here, since the behaviour is fairly predictable, and
+  // it's nice to save us a handful of string allocations if we only
+  // want one or the other
+  override lazy val path = s"/$group/$version"
```

Comment:
This isn't in the hot path so allocations of a few strings won't matter.  Begone, fowl lazy keyword!